### PR TITLE
Create isChainValid() function to easily test property chains

### PR DESCRIPTION
**Pull request by 1hitsong**
## Changes
Creates a new helper function to easily check if deeply buried properties are valid.

Backports changes from #1691 to fix unit tests in bug branch.

Backports `isAllValid()` from master to use in bug fixes

## Examples
`isChainValid(m.global, "session.user.settings.lastRunVersion")` returns true

`isChainValid(m.global, "session.nouser.settings.lastRunVersion")` returns false

### CODE REVIEW

File: source/utils/misc.bs [hunk 1]
```diff
@@ -299,6 +299,40 @@ function isValid(input as dynamic) as boolean
...
+
+' isChainValid: Returns whether or not all the properties in the passed property chain are valid.
+' Stops evaluating at first found false value
+'
+' @param {dynamic} root - high-level object to test property chain against
+' @param {string} propertyPath - chain of properties under root object to test
+' @return {boolean} indicating if all properties in chain are valid
+function isChainValid(root as dynamic, propertyPath as string) as boolean
+    rootPath = root
+    properties = propertyPath.Split(".")
```
Comment (neilsb) [inline source/utils/misc.bs:318]: Since you're testing if root is Valid, might be worth testing propertyPath is Valid too before calling Split on it?

Comment (1hitsong) [reply source/utils/misc.bs:318]: If propertyPath is invalid, it'll fail in the param casting as a string before we have a chance to test it.

I did, however, add 2 checks for propertyPath split - 1 for the count of the split array and 1 to confirm the first property isn't ""

image

**Review by neilsb — APPROVED** [review]

**Comment by cewert** [comment]
Can we add some unit tests for the new functions? Just to be sure they won't blow up when we start using them everywhere

**Comment by 1hitsong** [comment]
> Can we add some unit tests for the new functions? Just to be sure they won't blow up when we start using them everywhere

Tests for `chainIsValid()` and `isAllValid()` added.

**Review by cewert — APPROVED** [review]
```
Total: 105
   Passed: 105
   Crashed: 0
   Failed: 0
   Ignored: 0
   Time: 877ms
 
 
 RESULT: Success
```

**Pull request merged**